### ISSUE TITLE
pkg provides returns error 64 on any subcommand

### ISSUE THREAD

**Issue by dmikushin** [rosorio/pkg-provides#9]
Hi, thank you for such a nice plugin.

```
> uname -a
FreeBSD generic 13.2-STABLE FreeBSD 13.2-STABLE #0 stable/13-n254729-3912f99ecae6: Thu Mar  2 03:55:31 UTC 2023     root@releng3.nyi.freebsd.org:/usr/obj/usr/src/arm64.aarch64/sys/GENERIC arm64
> pkg plugins
NAME       DESC                                          VERSION   
provides   A plugin for querying which package provides a particular file 0.7.2
```

Whatever I try to run, pkg provides always returns error code 64. Maybe I have forgotten something really basic?

```
truss pkg provides -u
...
sigaction(SIGPIPE,{ SIG_IGN SA_RESTART ss_t },{ SIG_DFL 0x0 ss_t }) = 0 (0x0)
sigprocmask(SIG_SETMASK,{ },0x0)		 = 0 (0x0)
close(-1)					 ERR#9 'Bad file descriptor'
openat(AT_FDCWD,"/dev/null",O_RDWR,00)		 = 3 (0x3)
umask(0x12)					 = 18 (0x12)
fork()						 = 2365 (0x93d)
usage: pkg provides [-uf] [-r repo] pattern

A plugin for querying which package provides a particular file
wait4(2365,{ EXITED,val=64 },0x0,0x0)		 = 2365 (0x93d)
exit(0x40)					
process exit, rval = 64
```

**Comment by rosorio** [comment]
Hi,
Yes I'm aware of this problem who was reported a couple of weeks ago,
I investigate a bit and saw that the way packages ara currently compiled
for arm64, the getopt command is handle the wrong way.

I'm able to reproduce the problem on poudriere, but I have issues with the
emulation mode, so I'm wasn't able to explore the problem.

As I said in the PR currently open, you can easily fix the issue by rebuilding
provides plugin in your target environment, there is only one dependency
to install.

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=269567

**Comment by dmikushin** [comment]
Hi @rosorio , I'm sorry that I missed an existing report, and thank you for the detailed explanation.

To help you with investigation: is it possible to produce debug-enabled build in qemu environment, so that I can run it in gdb and tell you where exactly getopt is misbehaving?

**Comment by rosorio** [comment]
I want to thanks you for the heads-up about this issue, I finally figured out what the problem was.
As described in the PR, during last update I forgot to update the github hash, do the fix done un 0.7.3
never touch the FreeBSD ports.

I commit a fix in master and 2023Q1 branch, so if you are not building your own ports,
a new version 0.7.3_1 will be available in pkg mirrors in a couple of days.
